pages/items/[id]/index.js

```diff
@@ -32,7 +26,7 @@ export default function DetailsPage() {
     return <h2>{error}</h2>;
   }
 
-  return (
+  return id ? (
```

Comment:
This still looks odd to me.
Suggestion:
`if(!id) {return <h2>Loading...<h2>}` // returns loading if router is not ready to show id
`return <ItemDetails... ` // else returns ItemDetails